src/App.js

```diff
@@ -1,23 +1,51 @@
-import logo from './logo.svg';
-import './App.css';
+import axios from 'axios';
+import React, {useCallback, useEffect, useState} from "react";
 
 function App() {
+
+  const [data, setData] = useState({})
+  const [city, setCity] = useState("")
+  //const [ex, setEx] = useState({})
+  const [url, setUrl] = useState("")
+
+  const fetchCity = useCallback(() =>{
+    url !== "" &&
+    axios(url)
+        .then(function(response){
+          setData(response.data)
+        })
+  }, [url])
+
+  useEffect(() =>{
+    setUrl(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=a32a825225d19883c0b3eb8e44dd36a1`)
+  }, [city])
+
+
+  //function handleChange({target}) {
+  //  const {value, name} = target
+        //  setEx({
+  //    [name]: value
+        //  })
+  //}
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <form onSubmit={(e) => {
+        e.preventDefault()
+        fetchCity()}}>
+        <label>Nom :</label>
+        <input type="text" name="city" onChange={({target}) =>{
```

Comment:
Atomic design ? You should separate all your components in many atoms (input > form > view)